src/graphcast_agent/waku_handling.rs

```diff
@@ -116,14 +107,14 @@ pub fn unsubscribe_peer(
     graphcast_topic: &WakuPubSubTopic,
     content_topics: &[WakuContentTopic],
 ) -> Result<(), WakuHandlingError> {
-    let subscription: FilterSubscription =
-        content_filter_subscription(graphcast_topic, content_topics);
-    debug!(
-        peers = tracing::field::debug(&subscription),
-        "Unsubscribe content topics on filter protocol",
-    );
+    let content_filter =
+        ContentFilter::new(Some(graphcast_topic.to_string()), content_topics.to_vec());
     node_handle
-        .filter_unsubscribe(&subscription, Duration::new(6000, 0))
+        .filter_unsubscribe(
+            &content_filter,
+            node_handle.peer_id().unwrap(),
```

Comment:
Would you please take a look at the function signature for `filter_unsubscribe`?
From what I can see in go waku, "peerID should contain the ID of a peer this client is subscribed to."